### PR TITLE
Reduced logging

### PR DESCRIPTION
**Pull request by hackaugusto**
Merge after https://github.com/aiven/karapace/pull/391 (It fix port allocation for integration tests)

## Why?

Logging schemas has two problems (1) It is cpu intensive, and this is a problem while starting Karapace against a topic with a couple of thousands of schemas (2) Schemas are private information, and this changes the logging with them to `TRACE` level instead of `DEBUG`.

This also makes some logger instances private.

### CODE REVIEW

File: karapace/schema_reader.py
Comment (hackaugusto) [inline karapace/schema_reader.py:282]: I thought about not logging `subject`s too, but I think in that case we would have too little logging available.

File: karapace/schema_reader.py
Comment (tvainika) [inline karapace/schema_reader.py:253]: I'm not really fond of this non-standard `TRACE` level logging. I think we should just go with debug level logging for these, which is kind of standard, but not for production usage.

File: karapace/utils.py
Comment (tvainika) [inline karapace/utils.py:146]: Here also IMHO going backwards.

File: karapace/schema_reader.py
Comment (juha-aiven) [inline karapace/schema_reader.py:252]: Is the idea to hide the values or to have less logging?

If the latter, would it be too much to log `msg.key`. Now this message has no context of what's wrong. We could always truncate the key to e.g. 16 characters.

**Review by tvainika — CHANGES_REQUESTED** [review]
Seems I questioned some of your decision here. Maybe some discussion to have as we might just have differing opinions here?

File: karapace/schema_reader.py
Comment (juha-aiven) [inline karapace/schema_reader.py:261]: The same as above, maybe log the key or parts of it.

File: karapace/schema_reader.py
Comment (juha-aiven) [inline karapace/schema_reader.py:266]: Has this message been intentionally removed? Maybe combine these logged values the the above logging statement.

File: karapace/schema_reader.py
Comment (juha-aiven) [inline karapace/schema_reader.py:366]: Particular reason why the location of this logging has been changed?

File: karapace/schema_reader.py
Comment (juha-aiven) [inline karapace/schema_reader.py:371]: Any reason why before `typed_schema` was logged and now `schema`? Maybe because `schema` has more info?

File: karapace/statsd.py
Comment (juha-aiven) [inline karapace/statsd.py:129]: Why not log the `Exception`?

File: karapace/compatibility/jsonschema/checks.py
Comment (juha-aiven) [inline karapace/compatibility/jsonschema/checks.py:251]: Is this really unnecessary? Maybe leave, but without parameters. It doesn't look like something that would fill up logs.

**Comment by juha-aiven** [comment]
The PR is so big that it'd be worth mentioning in the description that it does other things too than just those mentioned, e.g. it makes logger instances private.

File: karapace/config.py
Comment (hackaugusto) [reply karapace/config.py:54]: what do you mean? `if 'password' in key'` or something like that?

I don't see the point of using a constant that looks out-of-place being bad at all. Sensitive data is usually the exception, and having it look weird can be a good thing to caught people's attention.

File: karapace/karapace.py
Comment (hackaugusto) [reply karapace/karapace.py:24]: Yep, that was intentional. This is not a stable api, the change doesn't make it harder to consume logs, and the existing code was super convoluted, where this log instance was overwriting the one from RestApp, and then both `KarapaceAll` and `KafkaRest` overwrote it too, but for some reason `KarapaceSchemaRegistry` did not. So in the end the name of the logger would change based on the flags provided in the configuration file, which doesn't seem to be useful for anything.

File: karapace/client.py
Comment (hackaugusto) [reply karapace/client.py:77]: would mentioning it make the commit any better? I'm adding it there anyways but the change is changing `log.info` to `self._log.error` and the text will be `changed log from info to error`

File: karapace/karapace_all.py
Comment (hackaugusto) [reply karapace/karapace_all.py:18]: same as https://github.com/aiven/karapace/pull/384#discussion_r844307880

File: karapace/master_coordinator.py
Comment (hackaugusto) [reply karapace/master_coordinator.py:122]: honestly I don't even remember anymore, gut feeling says the code was right.

File: karapace/schema_reader.py
Comment (hackaugusto) [reply karapace/schema_reader.py:252]: > Is the idea to hide the values or to have less logging?

I removed anything that is private data from the default log level, also the data you asked for is in the next line when verbose logging is used.

> Now this message has no context of what's wrong

That is what the exception message is for:

```
>>> import ujson
>>> import logging
>>> try:
...   ujson.loads('this is a test')
... except:
...   logging.exception('invalid json')
... 
ERROR:root:invalid json
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: Unexpected character found when decoding 'true'
```

File: karapace/schema_reader.py
Comment (hackaugusto) [reply karapace/schema_reader.py:261]: same answer, private data is removed. debugging data is available in the next line. context is available in the exception message.

File: karapace/schema_reader.py
Comment (hackaugusto) [reply karapace/schema_reader.py:266]: yes, this was intentionally removed, the previous line was the dispatch to the handlers, they extensive logging with additional context based on the message type

File: karapace/schema_reader.py
Comment (hackaugusto) [reply karapace/schema_reader.py:366]: because of what you said https://github.com/aiven/karapace/pull/384#discussion_r842390928

File: karapace/schema_reader.py
Comment (hackaugusto) [reply karapace/schema_reader.py:371]: more info

File: karapace/statsd.py
Comment (hackaugusto) [reply karapace/statsd.py:129]: the change is from `error` to `exception`. That is called `exception` because the exception is included.

File: karapace/schema_reader.py
Comment (hackaugusto) [reply karapace/schema_reader.py:253]: Any other arguments? That was intentional as trace is more verbose than debug.

File: karapace/schema_registry_apis.py
Comment (hackaugusto) [reply karapace/schema_registry_apis.py:647]: I thought that was easy enough to review because of the size. Here is the new PR: https://github.com/aiven/karapace/pull/385

File: karapace/compatibility/jsonschema/checks.py
Comment (hackaugusto) [reply karapace/compatibility/jsonschema/checks.py:251]: - This is a recursive call, number of log lines is undetermined
- The same information is encoded in the object instantiate in the next line with additional context (namely where in the recursion is the incompatibility), and that is logged by the caller of this function

**Comment by hackaugusto** [comment]
> The PR is so big that it'd be worth mentioning in the description that it does other things too than just those mentioned, e.g. it makes logger instances private.

Added `makes logger instances private` to the PR description

File: karapace/kafka_rest_apis/admin.py
Comment (jjaakola-aiven) [reply karapace/kafka_rest_apis/admin.py:16]: Maybe the logging could be removed in this file, it is used only to debug log retries.

File: karapace/kafka_rest_apis/consumer_manager.py
Comment (jjaakola-aiven) [inline karapace/kafka_rest_apis/consumer_manager.py:40]: Is this debug log line necessary?

File: karapace/kafka_rest_apis/consumer_manager.py
Comment (jjaakola-aiven) [inline karapace/kafka_rest_apis/consumer_manager.py:158]: I would combine the return value of `new_name` to the log. Also consider if this is debug log instead of info.
The group and name is logged as info also on the line 173.

File: karapace/kafka_rest_apis/consumer_manager.py
Comment (jjaakola-aiven) [inline karapace/kafka_rest_apis/consumer_manager.py:163]: This would be a `warn` instead of error.

**Review by jjaakola-aiven — COMMENTED** [review]
IMO the debug log calls could be removed totally.
Info logs also do much of tracing of the normal operation flow and I propose that the default log level should be set to `WARNING`.

File: karapace/kafka_rest_apis/admin.py
Comment (hackaugusto) [reply karapace/kafka_rest_apis/admin.py]: moved all the loggers to the module.

File: karapace/schema_reader.py
Comment (hackaugusto) [reply karapace/schema_reader.py]: Removed all TRACE calls.

File: karapace/utils.py
Comment (hackaugusto) [reply karapace/utils.py]: All loggers are now module scoped.

File: karapace/client.py
Comment (hackaugusto) [reply karapace/client.py]: Most modules are now at the module level.

File: karapace/config.py
Comment (tvainika) [reply karapace/config.py:56]: Something like that was my point, but resolving this as matter of style, and differing opinions, not rules

File: karapace/config.py
Comment (hackaugusto) [reply karapace/config.py:114]: For bug reports?

File: karapace/kafka_rest_apis/consumer_manager.py
Comment (hackaugusto) [reply karapace/kafka_rest_apis/consumer_manager.py]: done

File: karapace/config.py
Comment (jjaakola-aiven) [reply karapace/config.py]: Yes, and in my opinion info level is the correct for printing the startup configuration.

**Review by jjaakola-aiven — COMMENTED** [review]
LGTM

**Review by tvainika — APPROVED** [review]
LGTM, but pending merge until another pr merged as per pr description

**Pull request merged**
